internal/agent/upgrade.go

```diff
@@ -148,17 +95,133 @@ func determineUpgradeImage(version string) (string, error) {
 		Version: version,
 	})
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if img != "" {
-		return img, nil
+		return nil, nil
```

Comment:
is this really meant to return nil, nil?

Reply:
Oops! No this should have been : 

```
	if img != "" {
		return v1.NewSrcFromURI(img)
	}
```
fixed now